shub/utils.py

```diff
@@ -80,6 +82,25 @@ def _is_deploy_successful(last_logs):
         pass
 
 
+@contextlib.contextmanager
+def patch_sys_executable():
+    """
+    Context manager that monkey-patches sys.executable to point to the Python
+    interpreter.
+
+    Some scripts, in particular pip, depend on sys.executable pointing to the
+    Python interpreter. When frozen, however, sys.executable points to the
+    stand-alone file (i.e. the frozen script).
+    """
+    if getattr(sys, 'frozen', False):
+        orig_exe = sys.executable
+        sys.executable = find_exe('python')
```

Comment:
Indeed. It's a bit tricky because the Windows binary is always called `python.exe` (with no symlink from `python2`).

We could parse the output of `python --version` I guess.

Again this would be much nicer if we found a way to call a new Python instance from PyInstaller